File: readability.py
from cs50 import get_string

# Prompt user to input text

text = get_string("Text: ")

# Set variables for counters to appropriate values

letters = 0
words = 1
sentences = 0

# Calculate letter count

for i in text:
    if i.isalpha():
        letters += 1

# Calculate word count

    elif i == " ":
        words += 1

# Calculate sentence count

    elif i == "." or i == "?" or i == "!":
        sentences += 1

# Calculate Coleman-Liau Index

index = 0.0588 * (letters/words * 100) - 0.296 * (sentences/words * 100) - 15.8

# Return grade-level of text

if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print("Grade ", round(index))

File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{

//Getting User Input

    string text = get_string("Text: ");

    int letters = 0;
    int words = 1;
    int sentences = 0;

    for (int i = 0; i < strlen(text); i++)
    {

//Calculating Letters

        if (isalpha(text[i]))
        {
            letters++;
        }

//Calculating Words

        else if (text[i] == ' ')
        {
            words++;
        }

//Calculating Sentences

        else if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }

//Converting ints to floats so they compute in the equation

    float L = (float) letters / (float) words * 100;
    float S = (float) sentences / (float) words * 100;

//Coleman-Liau index

    int index = round(0.0588 * L - 0.296 * S - 15.8);

// Output of grade-level

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }

}